src/launch/mod.rs

```diff
@@ -261,3 +263,50 @@ pub struct TdxCapabilities {
 
     pub cpuid_configs: Vec<CpuidConfig>,
 }
+
+/// Manually create the wrapper for KVM_MEMORY_ENCRYPT_OP since `kvm_ioctls` doesn't
+/// support `.encrypt_op` for vcpu fds
+use vmm_sys_util::*;
+ioctl_iowr_nr!(
+    KVM_MEMORY_ENCRYPT_OP,
+    kvm_bindings::KVMIO,
+    0xba,
+    std::os::raw::c_ulong
+);
+
+pub struct TdxVcpu {
+    pub fd: kvm_ioctls::VcpuFd,
+}
+
+impl TdxVcpu {
+    pub fn new(fd: kvm_ioctls::VcpuFd) -> Result<TdxVcpu, TdxError> {
```

Comment:
A function named `new` usually indicates that their is some creation going on. In this instance, I would expect a `new` function to create the `VcpuFd` (keep in mind I don't think it should do this, just that the name seemingly suggests it will).

Why don't we encapsulate this in a:

`impl TryFrom<kvm_ioctls::VcpuFd> for TdxVcpu`?

Reply:
That's a good point, thanks for pointing that out. Fixed.